### PR TITLE
Choropleth text

### LINKED ISSUE #892: choropleth text controls
`choropleth` supports `text` but the widget isn't showing up. The easy fix is to add `choropleth` to the whitelist here https://github.com/plotly/react-chart-editor/blob/master/src/default_panels/StyleTracesPanel.js#L460 but this leads me to question why this is in a `TraceTypeSection` at all... what's the downside to just having this be a `Section`? Can someone dig into this please? :)

### LINKED ISSUE #901: hoverlabel.align attr: auto/left/right
New in 1.47.0 and needs to be added please!

### PR DESCRIPTION
**Pull request by dmt0**
Closes #892
Closes #901

### CODE REVIEW

File: dev/mocks.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   "/percy/waterfall.json",
   "/percy/sunburst.json",
   "/percy/sankey.json",
+  "/percy/choropleth.json",
```
Comment (VeraZab) [inline dev/mocks.json]: 👍

**Comment by VeraZab** [comment]
💃

File: src/default_panels/StyleLayoutPanel.js [hunk 2]
```diff
@@ -144,6 +144,16 @@ const StyleLayoutPanel = (props, {localize: _}) => (
       </PlotlySection>
       <PlotlySection name={_('Hover')}>
         <HovermodeDropdown label={_('Mode')} attr="hovermode">
+          <Dropdown
```
Comment (nicolaskruchten) [inline src/default_panels/StyleLayoutPanel.js]: this should be in Style/Traces, sorry, I don't think there's a global version

Comment (nicolaskruchten) [reply src/default_panels/StyleLayoutPanel.js]: ah never mind, there IS. but it should also be in Style/Traces (I think it's there also, right?)

File: src/default_panels/StyleTracesPanel.js [hunk 3]
```diff
@@ -477,21 +488,19 @@ const StyleTracesPanel = (props, {localize: _}) => (
...
-      ]}
+      traceTypes={allTraceTypes.filter(
+        t =>
+          ![
+            'histogram2d',
+            'histogram2dcontour',
+            'parcoords',
+            'parcats',
+            'sankey',
+            'table',
```
Comment (nicolaskruchten) [inline src/default_panels/StyleTracesPanel.js]: do Violin, Box and Histogram support Text??

File: src/lib/constants.js [hunk 4]
```diff
@@ -142,17 +142,4 @@ export const COLORS = {
...
-  COLORS.cookedAsparagusGreen,
-  COLORS.brickRed,
-  COLORS.mutedPurple,
-  COLORS.chestnutBrown,
-  COLORS.raspberryYogurtPink,
-  COLORS.middleGray,
-  COLORS.curryYellowGreen,
-  COLORS.blueTeal,
-];
+export const DEFAULT_COLORS = Object.values(COLORS);
```
Comment (nicolaskruchten) [inline src/lib/constants.js]: where do we even use these default colours?

File: src/default_panels/StyleTracesPanel.js (hunk 3 above)
Comment (dmt0) [reply src/default_panels/StyleTracesPanel.js]: image

File: src/lib/constants.js (hunk 4 above)
Comment (dmt0) [reply src/lib/constants.js]: We don't, but we do export them - so that's part of the API I guess.

Comment (nicolaskruchten) [reply src/lib/constants.js]: bah. ok :)

File: src/default_panels/StyleTracesPanel.js (hunk 3 above)
Comment (nicolaskruchten) [reply src/default_panels/StyleTracesPanel.js]: whaddaya know. ok then

File: src/default_panels/StyleLayoutPanel.js (hunk 2 above)
Comment (dmt0) [reply src/default_panels/StyleLayoutPanel.js]: Done

**Comment by nicolaskruchten** [comment]
✨ ... merging!

**Pull request merged**